example/ansi_code_example.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:math';
+import 'package:io/src/ansi_code.dart';
+
+/// Prints a sample of all of the `AnsiCode` values.
+void main() {
+  if (!stdout.supportsAnsiEscapes) {
+    print('`stdout.supportsAnsiEscapes` is `false`.');
+    print("Don't expect pretty output.");
+  }
+  _preview('Foreground', allForegrounds);
+  _preview('Background', allBackgrounds);
+  _preview('Styles', allStyles);
+}
+
+void _preview(String name, List<AnsiCode> values) {
+  print('');
+  final longest =
+      values.fold<int>(0, (len, code) => max(len, code.name.length));
+
+  void _print(AnsiCode code) {
```

Comment:
[nit] does this need to be defined here or can it be top level? Maybe even skip the method and put these 2 lines in the loop body?